src/creeps/Zombie.cpp

```diff
@@ -1,61 +1,307 @@
+                c = '0' + (j + DIR_CAP / 2) % DIR_CAP;
+                path = c + path;
+                x += MX[j];
+                y += MY[j];
+            }
+
+            // empty the leftover nodes
+            pq[index] = priority_queue<Node>();
+
+            setPath(path);
+            return path;
+        }
+
+        // traverse neighbors
+        for (i = 0; i < DIR_CAP;i++) {
+            // neighbor coordinates
+            xdx = x + MX[i];
+            ydy = y + MY[i];
+
+            // not evaluated & not outside (bound checking)
+            if (!(xdx < 0 || xdx > COL -1 || ydy < 0 || ydy > ROW - 1
+                    || gameMap[xdx][ydy] == 1 || closedNodes[xdx][ydy] == 1)) {
+
+                // generate a child node
+                childNode = Node(xdx, ydy, curNode.getLevel(), curNode.getPriority());
+                childNode.nextLevel(i);
+                childNode.updatePriority(xNodeDest, yNodeDest);
+
+                // if it is not in the open list then add into that
+                if (openNodes[xdx][ydy] == 0) {
+                    openNodes[xdx][ydy] = childNode.getPriority();
+                    pq[index].push(childNode);
+                    // update the parent direction info
+                    dirMap[xdx][ydy] = (i + DIR_CAP/2)%DIR_CAP;
+                } else if (openNodes[xdx][ydy] > childNode.getPriority()) {
+                    // update the priority info
+                    openNodes[xdx][ydy]= childNode.getPriority();
+                    // update the parent direction info
+                    dirMap[xdx][ydy] = (i + DIR_CAP/2)%DIR_CAP;
+
+                    // use a queue and a backup queue to put the best node (with highest priority)
+                    // on the top of the queue, which can be chosen later on to build the path.
+                    while (!(pq[index].top().getXPos() == xdx &&
+                           pq[index].top().getYPos() == ydy)) {
+                        pq[1-index].push(pq[index].top());
+                        pq[index].pop();
+                    }
+
+                    pq[index].pop();
+
+                    // switch to pq with smaller size
+                    if (pq[index].size() > pq[1-index].size()) {
+                        index = 1 - index;
+                    }
+
+                    while (!pq[index].empty()) {
+                        pq[1-index].push(pq[index].top());
+                        pq[index].pop();
+                    }
+                    index = 1 - index;
+                    pq[index].push(childNode);
+                }
+            }
+        }
     }
-    movementHitBox->move(getX(),getY());
-    projectileHitBox->move(getX(),getY());
 
-    setDX(x*cosVal);
-    setDY(y*sinVal);
- }
+    return ""; // no route found
+}
+
+/**
+ * Check to see if the zombie is still within the screen bounds before moving
+ * Fred Yang
+ * Feb 14
+ */
+bool Zombie::checkBounds(const float x, const float y) const {
```

Comment:
Then make it static
This method doesn't refer to any members of zombie, so it can be static
static constexpr method should hopefully fix things